sdk/storage/Azure.Storage.DataMovement/src/UriToStreamJobPart.cs

```diff
@@ -237,34 +237,34 @@ internal async Task UnknownDownloadInternal()
                 }
                 // If the initial request returned no content (i.e., a 304),
                 // we'll pass that back to the user immediately
-                long initialLength = initialResult.Properties.ContentLength;
+                long? initialLength = initialResult?.Range.Length;
 
                 // There needs to be at least 1 chunk to create the blob even if the
                 // length is 0 bytes.
-                if (initialResult == default || initialLength == 0)
+                if (initialResult == default || initialLength.GetValueOrDefault() == 0)
```

Comment:
I'm a bit confused on these changes. What would it mean for `initialLength` to be null? The default value of `initialLength` would be `null` right, not `0`? If so, `initialLength.GetValueOrDefault() == 0` doesn't work right? But after this check you start using `nitialLength.Value` which will throw if its null.

Do we need a special check for null/default here. Maybe we can somehow still save `initialLength` as non-nullable with some checks?

Reply:
So I made this change, because of the line above change that I made, to grab the length of range and not the length of resource itself.

It could be default because an empty blob or file would have no range.

Let me try adjust the line above and see if this helps. I think I shouldn't have done the range thing and grabbed the length of the storage resource itself.

Reply:
Ah I was also under the impression that `GetValueOrDefault` was like the equivalent of doing `if (length ?? 0) == 0`. Will change it to that, to be clearer.